platform-tests/src/test/java/org/junit/platform/engine/support/descriptor/MethodSourceTests.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.platform.engine.support.descriptor;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.lang.reflect.Method;
+
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.util.PreconditionViolationException;
+
+class MethodSourceTests {
+
+	@Test
+	void instantiatingWithNullNamesShouldThrowPreconditionViolationException() {
+		assertThrows(PreconditionViolationException.class, () -> new MethodSource(null, null));
+	}
+
+	@Test
+	void instantiatingWithBlankNamesShouldThrowPreconditionViolationException() {
```

Comment:
Those are technically _empty_ names.

Please implement additional tests for _blank_ names.